src/_imaging.c

```diff
@@ -1146,11 +1146,15 @@ static inline int
 _getxy(PyObject *xy, int *x, int *y) {
     PyObject *value;
 
-    if (!PyTuple_Check(xy) || PyTuple_GET_SIZE(xy) != 2) {
+    int tuple = PyTuple_Check(xy);
```

Comment:
So many checks here? Isn't a way do something like `PyArg_ParseTuple(xy, "ii", x, y)`?

Reply:
Possibly - but if your goal is simplicity, then I should mention that the simplest solution is to implement this change in Python instead of C.
```diff
diff --git a/src/PIL/Image.py b/src/PIL/Image.py
index 476ed0122..6ea711b56 100644
--- a/src/PIL/Image.py
+++ b/src/PIL/Image.py
@@ -1565,7 +1565,7 @@ class Image:
         self.load()
         if self.pyaccess:
             return self.pyaccess.getpixel(xy)
-        return self.im.getpixel(xy)
+        return self.im.getpixel(tuple(xy))
 
     def getprojection(self):
         """
```

Reply:
My original intention here was to make this change close to the source of the error though. If the change is being made higher up, then my argument that it would be awkward to update the error message for just this scenario falls apart, since you could just add a new error message.

Then I feel less sure about this change, as the discourse in this issue doesn't have a clear conclusion, and I'm reminded of #3738, in particular https://github.com/python-pillow/Pillow/issues/3738#issuecomment-476701199 and https://github.com/python-pillow/Pillow/issues/3738#issuecomment-507126728.

https://github.com/python-pillow/Pillow/issues/3738#issuecomment-507126728
> Also, tuples and lists are for slightly different things (immutable/mutable, structure/order). Does that really matter? I think maybe it does: semantically you know a coordinate like (100, 200) can't have a third value; but [100, 200] looks like it could. So I'm tending towards keeping it as is.

Reply:
I didn't have strong feelings in https://github.com/python-pillow/Pillow/issues/3738#issuecomment-507126728, but in the years since then am now tending the other way, maybe we could accept more iterables for these kinds of things?

Reply:
Ok, I've pushed a commit with a simpler change in Python.